libpod/container_internal_linux.go

```diff
@@ -827,17 +820,16 @@ func (c *Container) generatePasswd() (string, error) {
 		return "", nil
 	}
 	if groupspec != "" {
-		if !c.state.Mounted {
-			return "", errors.Wrapf(ErrCtrStateInvalid, "container %s must be mounted in order to translate group field for passwd record", c.ID())
-		}
-		group, err = lookup.GetGroup(c.state.Mountpoint, groupspec)
-		if err != nil {
-			if err == user.ErrNoGroupEntries {
+		ugid, err := strconv.ParseUint(groupspec, 10, 32)
+		if err == nil {
+			gid = int(ugid)
+		} else {
+			group, err := lookup.GetGroup(c.state.Mountpoint, groupspec)
```

Comment:
Should this be patched into lookup.GetGroup(), instead of working around it here? We use GetGroup in Buildah, so if possible we'd like to share the fix with them

Reply:
> Should this be patched into `lookup.GetGroup()`, instead of working around it here?

Well, there's also `Group.Name`, `Group.Pass` and `Group.List` in `user.Group` that we'd be unable to populate.  So what I'm doing here (not hitting the container filesystem at all, because we only need the GID) doesn't seem like a good fit for `GetGroup`.

But you could have `GetGroup` return a `*user.Group` with only `Gid` set and still return `ErrNoGroupEntries` so the caller that wanted those other fields but could limp by with just the GID could just ignore `ErrNoGroupEntries` instead of performing their own integer check.  Should I add a commit to this PR to make that `GetGroup` change?

Reply:
Yes lets do that

Reply:
> Yes lets do that

Ok, pushed 7fae5d5 with this logic.